plotting/auxiliars_withSF.py

```diff
@@ -0,0 +1,238 @@
+
+def getSF2El(lep1pt, lep1eta, lep2pt, lep2eta, year, var, mode  = 0):
+  ceval = CorrectionSet.from_file("C:/Users/slatt/Downloads/POGJsons/masterJSON.json") #Load JSON file with Trigger Efficiencies
+
+  if lep1pt > 200.0: #No Trigger Scale Factor Applied (Out of bounds Pt)
+    return getSFEl(lep1pt, lep1eta, year, var)*getSFEl(lep2pt, lep2eta, year, var)
+
+  if mode == 0: #Pt-Eta Binning for Trigger Scale Factors:
+    if var > 0:
+      if ceval['UL-PtEta-Trigger-SFs'].evaluate(year,"sf-up","Electron",lep1eta, lep1pt) < 0.25: #Make sure uncertainty in bin is not too high
+        return getSFEl(lep1pt, lep1eta, year, var)*getSFEl(lep2pt, lep2eta, year, var)*(ceval['UL-PtEta-Trigger-SFs'].evaluate(year,"sf","Electron",lep1eta, lep1pt) + var * ceval['UL-PtEta-Trigger-SFs'].evaluate(year,"sf-up","Electron",lep1eta, lep1pt))
+    else:
+      if ceval['UL-PtEta-Trigger-SFs'].evaluate(year,"sf-down","Electron",lep1eta, lep1pt) < 0.25: #Make sure uncertainty in bin is not too high
+        return getSFEl(lep1pt, lep1eta, year, var)*getSFEl(lep2pt, lep2eta, year, var)*(ceval['UL-PtEta-Trigger-SFs'].evaluate(year,"sf","Electron",lep1eta, lep1pt) + var * ceval['UL-PtEta-Trigger-SFs'].evaluate(year,"sf-down","Electron",lep1eta, lep1pt))
+
+  if mode == 1: #Pt-Pt Binning for Trigger Scale Factors:
+    if var > 0:
+      if ceval['UL-PtPt-Trigger-SFs'].evaluate(year,"sf-up","Electron",lep2pt, lep1pt) < 0.25: #Make sure uncertainty in bin is not too high
+        return getSFEl(lep1pt, lep1eta, year, var)*getSFEl(lep2pt, lep2eta, year, var)*(ceval['UL-PtPt-Trigger-SFs'].evaluate(year,"sf","Electron",lep2pt, lep1pt) + var * ceval['UL-PtPt-Trigger-SFs'].evaluate(year,"sf-up","Electron",lep2pt, lep1pt))
+    else:
+      if ceval['UL-PtPt-Trigger-SFs'].evaluate(year,"sf-down","Electron",lep2pt, lep1pt) < 0.25: #Make sure uncertainty in bin is not too high
+        return getSFEl(lep1pt, lep1eta, year, var)*getSFEl(lep2pt, lep2eta, year, var)*(ceval['UL-PtPt-Trigger-SFs'].evaluate(year,"sf","Electron",lep2pt, lep1pt) + var * ceval['UL-PtPt-Trigger-SFs'].evaluate(year,"sf-down","Electron",lep2pt, lep1pt))
+
+  return getSFEl(lep1pt, lep1eta, year, var)*getSFEl(lep2pt, lep2eta, year, var)
+
+def getSF(nmu, lep1pt, lep1eta, lep2pt, lep2eta, year, var=0, mode = 0):
+  if nmu == 2:
+    var = 0 if abs(var) >= 2 else var
+    sf = getSF2Mu(lep1pt, lep1eta, lep2pt, lep2eta, year, var)
+  else:
+    var = var/2 if abs(var) >= 2 else 0
+    sf = getSF2El(lep1pt, lep1eta, lep2pt, lep2eta, year, var)
+  return sf
+
+# Mode = 0 for Pt-Eta Trigger SFs, Mode = 1 for Lead/Sublead Pt Trigger SFs
+def SF(x, year, var=0, mode = 0): # Var encodes the uncertainty-varied SF (+-1 for varying muons, +-2 for varying electrons)
+  w = np.array([])
+  for i in range(len(x)):
+    w = np.append(w, getSF(x["nmuons"][i], x["leadlep_pt"][i], x["leadlep_eta"][i], x["subleadlep_pt"][i], x["subleadlep_eta"][i], year, var))
```

Comment:
Is mode ever passed down to getSF in L208? With this code it appears that the "mode" argument in SF() never goes down as mode to getSF()